<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1135</width>
    <height>741</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>STA - Mass Spring Demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="s_mainWindowGridLayout">
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="displayControlsGB">
        <property name="title">
         <string>Display</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="4" column="0">
          <widget class="QCheckBox" name="displayShearSpringsCheckBox">
           <property name="text">
            <string>Shear Springs</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="displayStructuralSpringsCheckBox">
           <property name="text">
            <string>Structural Springs</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QCheckBox" name="displayBendSpringsCheckBox">
           <property name="text">
            <string>Bend Springs</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="displayMassesCheckBox">
           <property name="text">
            <string>Masses</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="displayMeshCheckBox">
           <property name="text">
            <string>Mesh</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="physicsControlsGB">
        <property name="title">
         <string>Controls</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="7" column="0">
          <widget class="QPushButton" name="resetJelloPushButton">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0" rowspan="2">
          <layout class="QFormLayout" name="formLayout">
           <item row="1" column="0">
            <widget class="QLabel" name="springConstantLabel">
             <property name="text">
              <string>Spring</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="springConstantSpinBox">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="dampingConstantLabel">
             <property name="text">
              <string>Damping</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="dampingConstantSpinBox">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>2.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="jelloMassLabel">
             <property name="text">
              <string>Jello Mass</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="jelloMassSpinBox">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="gravityLabel">
             <property name="text">
              <string>Gravity</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QDoubleSpinBox" name="gravitySpinBox">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="minimum">
              <double>-99.989999999999995</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>-9.810000000000000</double>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="timeStepLabel">
             <property name="text">
              <string>Timestep</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QDoubleSpinBox" name="timeStepSpinBox">
             <property name="decimals">
              <number>4</number>
             </property>
             <property name="singleStep">
              <double>0.001000000000000</double>
             </property>
             <property name="value">
              <double>0.010000000000000</double>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="subStepsLabel">
             <property name="text">
              <string>Substeps</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QSpinBox" name="subStepsSpinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>999</number>
             </property>
             <property name="value">
              <number>10</number>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="integratorComboBox"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="integratorLabel">
             <property name="text">
              <string>Integrator</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="frictionLabel">
             <property name="text">
              <string>Friction</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="frictionSpinBox">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>0.200000000000000</double>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="recoveryConstantLabel">
             <property name="text">
              <string>Recovery</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="recoverySpinBox">
             <property name="decimals">
              <number>3</number>
             </property>
             <property name="maximum">
              <double>1.000000000000000</double>
             </property>
             <property name="singleStep">
              <double>0.100000000000000</double>
             </property>
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="massCountLabel">
           <property name="text">
            <string>Mass Count: ?</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="springCountLabel">
           <property name="text">
            <string>Spring Count: ?</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QCheckBox" name="enableSphereColliderCheckBox">
           <property name="text">
            <string>Sphere Collider</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="framesPerSecondLabel">
           <property name="text">
            <string>FPS: ?</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1135</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
